8.0.0-beta1 release notes (#79969)
elastic/elasticsearch: docs/reference/release-notes/8.0.0-beta1.asciidoc

:es-issue: https://github.com/elastic/elasticsearch/issues/
:es-pull:  https://github.com/elastic/elasticsearch/pull/

[[release-notes-8.0.0-beta1]]
== {es} version 8.0.0-beta1

> 

Reviewer: ```suggestion
beta::[This functionality is in alpha and is subject to change. The design and code is less mature than official GA features and is being provided as-is with no warranties. Alpha features are not subject to the support SLA of official GA features.]

```

> Also see <<breaking-changes-8.0,Breaking changes in 8.0>>.

Reviewer: ```suggestion
coming::[8.0.0-beta1]

Also see <<breaking-changes-8.0,Breaking changes in 8.0>>.
```

[[known-issues-8.0.0-beta1]]
[float]
=== Known issues

> * If you are using {ml}, it is not safe to perform a rolling upgrade from 7.16.x 
>   to 8.0.0-beta1. It may lead to cluster state propagation issues in the mixed 
>   version cluster. It is recommended waiting for 8.0.0-rc1 before testing a 
>   rolling upgrade from 7.16.x. Wait for GA before attempting to upgrade any 
>   production cluster to version 8.

Reviewer: ```suggestion
* If you're using {ml}, it's not safe to perform a rolling upgrade from `7.16.x `
  to `8.0.0-beta1`. Doing so might lead to cluster state propagation issues in the mixed 
  version cluster. We recommend waiting for `8.0.0-rc1` before testing a 
  rolling upgrade from `7.16.x`. Do not attempt to upgrade any 
  production cluster to pre-release software.
```

Reviewer: cc: @lcawl to lend her ML expertise

> * If you are using {ml}, it is not safe to upgrade to 8.0.0-beta1
>   if the cluster you are upgrading was first used prior to 7.7.0.
>   If you attempt such an upgrade the filtered aliases against
>   the {ml} results indices will lose their filters causing {ml}
>   anomaly detection results for different jobs to be mixed up.
>   Wait for 8.0.0-rc1 before attempting to upgrade any cluster
>   where {ml} was used prior to version 7.7.0 to version 8.

Reviewer: ```suggestion
* If you're using {ml}, it's not safe to upgrade to `8.0.0-beta1`
  if the cluster you're upgrading was first used prior to `7.7.0`.
  If you attempt such an upgrade the filtered aliases against
  the {ml} results indices will lose their filters, causing {ml}
  anomaly detection results for different jobs to be mixed.
  Wait for `8.0.0-rc1` before attempting to test upgrading a cluster
  where {ml} was used prior to version `7.7.0`.
```

Reviewer: cc: @lcawl to lend her ML expertise

[[breaking-8.0.0-beta1]]
[float]
=== Breaking changes

CRUD::

> * Return 200 OK response code for a cluster health timeout [OPEN] {es-pull}78968[#78968] (issues: {es-issue}70849[#70849], {es-issue}78180[#78180], {es-issue}78940[#78940])

Reviewer: ```suggestion
* Return 200 OK response code for a cluster health timeout {es-pull}78968[#78968] (issues: {es-issue}70849[#70849], {es-issue}78180[#78180], {es-issue}78940[#78940])
```

ILM+SLM::

> * Always enforce default tier preference (ENFORCE_DEFAULT_TIER_PREFERENCE is ignored) {es-pull}79751[#79751] (issue: {es-issue}76147[#76147])

Reviewer: ```suggestion
* Always enforce default tier preference {es-pull}79751[#79751] (issue: {es-issue}76147[#76147])
```

Infra/Core::
* Remove Joda dependency {es-pull}79007[#79007]
* Remove Joda support from date formatters {es-pull}78990[#78990]

Ingest::
* Remove default maxmind geoip databases from distribution {es-pull}78362[#78362] (issue: {es-issue}68920[#68920])

License::
* Enforce license expiration {es-pull}79671[#79671]

Packaging::
* Require Java 17 for running Elasticsearch {es-pull}79873[#79873]

[[deprecation-8.0.0-beta1]]
[float]
=== Deprecations

Authentication::
* Deprecate setup-passwords tool {es-pull}76902[#76902]

CRUD::
* Deprecate returning 408 for a server timeout on `_cluster/health` {es-pull}78180[#78180] (issue: {es-issue}70849[#70849])
* Remove `indices_segments` 'verbose' parameter {es-pull}78451[#78451] (issue: {es-issue}75955[#75955])

Monitoring::
* Add deprecation info API entries for deprecated monitoring settings {es-pull}78799[#78799]
* Automatically install monitoring templates at plugin initialization {es-pull}78350[#78350]
* Remove Monitoring ingest pipelines {es-pull}77459[#77459] (issue: {es-issue}50770[#50770])

[[feature-8.0.0-beta1]]
[float]
=== New features

Security::

> * Auto configure TLS for new nodes of new clusters {es-pull}77231[#77231] (issues: {es-issue}75144[#75144], {es-issue}75704[#75704])

Reviewer: ```suggestion
* Auto-configure TLS for new nodes of new clusters {es-pull}77231[#77231] (issues: {es-issue}75144[#75144], {es-issue}75704[#75704])
```

[[enhancement-8.0.0-beta1]]
[float]
=== Enhancements

Authentication::

> * CLI tool to (re)set users password {es-pull}79709[#79709]

Reviewer: ```suggestion
* New CLI tool to reset password for built-in users {es-pull}79709[#79709]
```

* Optimize source extraction for `categorize_text` aggregation {es-pull}79099[#79099]
* The Linux build platform for the {ml} C++ code is now CentOS 7 running gcc 10.3. {ml-pull}2028[#2028]

Mapping::
* Add support for configuring HNSW parameters {es-pull}79193[#79193] (issue: {es-issue}78473[#78473])
* Extend `dense_vector` to support indexing vectors {es-pull}78491[#78491] (issue: {es-issue}78473[#78473])

Monitoring::
* Add previously removed Monitoring settings back for 8.0 {es-pull}78784[#78784]
* Change Monitoring plugin cluster alerts to not install by default {es-pull}79657[#79657]

Packaging::
* Allow total memory to be overridden {es-pull}78750[#78750] (issue: {es-issue}65905[#65905])

Search::
* Node level can match action {es-pull}78765[#78765]
* TSDB: Add time series information to field caps {es-pull}78790[#78790] (issue: {es-issue}74660[#74660])
* Add new kNN search endpoint {es-pull}79013[#79013] (issue: {es-issue}78473[#78473])
* Disallow kNN searches on nested vector fields {es-pull}79403[#79403] (issue: {es-issue}78473[#78473])
* Ensure kNN search respects authorization {es-pull}79693[#79693] (issue: {es-issue}78473[#78473])

> * Load knn vectors format with mmapfs {es-pull}78724[#78724] (issue: {es-issue}78473[#78473])

Reviewer: ```suggestion
* Load kNN vectors format with mmapfs {es-pull}78724[#78724] (issue: {es-issue}78473[#78473])
```

Snapshot/Restore::
* Allow listing older repositories {es-pull}78244[#78244]
* Optimize SLM Policy Queries {es-pull}79341[#79341] (issue: {es-issue}79321[#79321])
* Upgrade repository-hdfs to Hadoop 3 {es-pull}76897[#76897]

Transform::
* Prevent old beta transforms from starting {es-pull}79712[#79712]

TSDB::
* Automatically add timestamp mapper {es-pull}79136[#79136]
* Create a coordinating node level reader for tsdb {es-pull}79197[#79197]
* Fix tsdb's shrink test in multi-version cluster [OPEN] {es-pull}79940[#79940] (issue: {es-issue}79936[#79936])

[[bug-8.0.0-beta1]]
[float]
=== Bug fixes

Infra/Settings::
* Stricter `UpdateSettingsRequest` parsing on the REST layer {es-pull}79227[#79227] (issue: {es-issue}29268[#29268])
* Set Auto expand replica on deprecation log data stream {es-pull}79226[#79226] (issue: {es-issue}78991[#78991])

Machine Learning::
* Add timeout parameter for delete trained models API {es-pull}79739[#79739] (issue: {es-issue}77070[#77070])
* Tone down ML unassigned job notifications {es-pull}79578[#79578] (issue: {es-issue}79270[#79270])
* Use a new annotations index for future annotations {es-pull}79006[#79006] (issue: {es-issue}78439[#78439])

Search::

> * CI fix - remove unsafe assertion in wildcard field {es-pull}78966[#78966]

Reviewer: ```suggestion
* Remove unsafe assertion in wildcard field {es-pull}78966[#78966]
```

[[regression-8.0.0-beta1]]
[float]
=== Regressions

Search::
* Disable numeric sort optimization conditionally {es-pull}78103[#78103]

[[upgrade-8.0.0-beta1]]
[float]
=== Upgrades

Search::

> * Update lucene 9 snapshot {es-pull}79701[#79701] {es-pull}79138[#79138] {es-pull}78548[#78548] {es-pull}78286[#78286] {es-pull}73324[#73324] {es-pull}79461[#79461]

Reviewer: ```suggestion
* Update Lucene 9 snapshot {es-pull}79701[#79701] {es-pull}79138[#79138] {es-pull}78548[#78548] {es-pull}78286[#78286] {es-pull}73324[#73324] {es-pull}79461[#79461]
```
